src/matching/Generic_vs_generic.ml

```diff
@@ -2510,6 +2517,26 @@ and m_stmt a b =
         ~else_:(fail ())
   (* equivalence: *)
   | G.ExprStmt (a1, _), B.Return (_, Some b1, _) -> m_expr_deep a1 b1
+  (* implicit return *)
+  (* We try to match the FBExpr cases below where function
```

Comment:
This stuff may fit better in `Matching_visitor`. cc @aryx

Reply:
I'll try it out a bit to see if it actually fits better there or not.

Reply:
Good callout. Added some changes in `Matching_visitor`.